### ISSUE TITLE
Can I assign private IP to VM in DevTest Labs?

### ISSUE THREAD

**Issue by Evit15** [Azure/azure-devtestlab#599]
Hi,

Can I use ARM template to assign Private IP when I create new VM in DevTest Lab?

**Comment by petehauge** [comment]
Yep!  I included a JSON template below for creating a VM with a private IP address.  In particular, this line: "disallowPublicIpAddress": true  .

Hope that helps!
~Pete

```json
{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "newVMName": {
      "type": "string",
    },
    "labName": {
      "type": "string",
    },
    "size": {
      "type": "string",
      "defaultValue": "Standard_D4s_v3"
    },
    "userName": {
      "type": "string",
    },
    "password": {
      "type": "securestring",
    }
  },
  "variables": {
    "labSubnetName": "[concat(variables('labVirtualNetworkName'), 'Subnet')]",
    "labVirtualNetworkId": "[resourceId('Microsoft.DevTestLab/labs/virtualnetworks', parameters('labName'), variables('labVirtualNetworkName'))]",
    "labVirtualNetworkName": "[concat('Dtl', parameters('labName'))]",
    "vmId": "[resourceId ('Microsoft.DevTestLab/labs/virtualmachines', parameters('labName'), parameters('newVMName'))]",
    "vmName": "[concat(parameters('labName'), '/', parameters('newVMName'))]"
  },
  "resources": [
    {
      "apiVersion": "2018-10-15-preview",
      "type": "Microsoft.DevTestLab/labs/virtualmachines",
      "name": "[variables('vmName')]",
      "location": "[resourceGroup().location]",
      "properties": {
        "labVirtualNetworkId": "[variables('labVirtualNetworkId')]",
        "notes": "Windows Server 2019 Datacenter",
        "galleryImageReference": {
          "offer": "WindowsServer",
          "publisher": "MicrosoftWindowsServer",
          "sku": "2019-Datacenter",
          "osType": "Windows",
          "version": "latest"
        },
        "size": "[parameters('size')]",
        "userName": "[parameters('userName')]",
        "password": "[parameters('password')]",
        "isAuthenticationWithSshKey": false,
        "labSubnetName": "[variables('labSubnetName')]",
        "disallowPublicIpAddress": true,
        "storageType": "StandardSSD",
        "allowClaim": false
      }
    }
  ],
  "outputs": {
    "labVMId": {
      "type": "string",
      "value": "[variables('vmId')]"
    }
  }
}
```

**Comment by Evit15** [comment]
Hi @petehauge 

Thanks for your reply!
But I want to make clear my question. I would like to assign specific private IP to when I create new VM in DevTest Lab.

This is my ARM template:

`{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "newVMName": {
      "type": "string",
      "defaultValue": "testVM"
    },
    "labName": {
      "type": "string",
      "defaultValue": "testLab"
    },
    "size": {
      "type": "string",
      "defaultValue": "Standard_B2s"
    },
    "userName": {
      "type": "string",
      "defaultValue": "test"
    },
    "password": {
      "type": "securestring",
      "defaultValue": "[[[test]]"
    }
  },
  "variables": {
    "labSubnetName": "[concat(variables('labVirtualNetworkName'), 'Subnet')]",
    "labVirtualNetworkId": "[resourceId('Microsoft.DevTestLab/labs/virtualnetworks', parameters('labName'), variables('labVirtualNetworkName'))]",
    "labVirtualNetworkName": "[concat('Dtl', parameters('labName'))]",
    "vmId": "[resourceId ('Microsoft.DevTestLab/labs/virtualmachines', parameters('labName'), parameters('newVMName'))]",
    "vmName": "[concat(parameters('labName'), '/', parameters('newVMName'))]"
  },
  "resources": [
    {
      "apiVersion": "2018-10-15-preview",
      "type": "Microsoft.DevTestLab/labs/virtualmachines",
      "name": "[variables('vmName')]",
      "location": "[resourceGroup().location]",
      "properties": {
        "labVirtualNetworkId": "[variables('labVirtualNetworkId')]",
        "notes": "Windows 10 Pro, Version 1903",
        "galleryImageReference": {
          "offer": "Windows-10",
          "publisher": "MicrosoftWindowsDesktop",
          "sku": "19h1-pro",
          "osType": "Windows",
          "version": "latest"
        },
        "size": "[parameters('size')]",
        "userName": "[parameters('userName')]",
        "password": "[parameters('password')]",
        "isAuthenticationWithSshKey": false,
        "artifacts": [
        ],
        "labSubnetName": "[variables('labSubnetName')]",
        "disallowPublicIpAddress": true,
        "storageType": "Standard",
        "allowClaim": false,
        "networkInterface": {
          "privateIPAddress": "10.0.0.6",
          "sharedPublicIpAddressConfiguration": {            
            "inboundNatRules": [
              {
                "transportProtocol": "tcp",
                "backendPort": 3389
              }
            ]
          }
        }
      }
    }
  ],
  "outputs": {
    "labVMId": {
      "type": "string",
      "value": "[variables('vmId')]"
    }
  }
}`

**Comment by petehauge** [comment]
Oh - I see!  Sorry for the confusion.  I took a look at the schema and it appears your template is correctly formatted, I just don't know if this is a supported feature in DevTest Labs.  Let's reopen the issue for one of the developers in this area to respond.  Thanks!
